File: backend/ngts/src/main/java/com/three/ngts/Entity/TicketAddress.java
package com.three.ngts.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "ticket_address")
public class TicketAddress {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ticket_address_id")
    private Long ticketAddressId;

    @ManyToOne
    @JoinColumn(name = "source", nullable = false)
    private District source;

    @ManyToOne
    @JoinColumn(name = "destination", nullable = false)
    private District destination;

    @ManyToOne
    @JoinColumn(name = "ticket_id", nullable = false)
    private Ticket ticket;
}

File: frontend/next-generation-ticketing-system/src/Entity/TicketAddress.ts
import District from "./District";
import Ticket from "./Ticket";

class TicketAddress {
  ticketAddressId: number;
  source: District;
  destination: District;
  ticket: Ticket;

  constructor(
    ticketAddressId: number,
    source: District,
    destination: District,
    ticket: Ticket
  ) {
    this.ticketAddressId = ticketAddressId;
    this.source = source;
    this.destination = destination;
    this.ticket = ticket;
  }
}

export default TicketAddress;
